components/table/table-wrapper.js

```diff
@@ -33,6 +38,55 @@ export const tableStyles = css`
 		text-align: right;
 	}
 
+	.d2l-table > * > tr > :has(.d2l-checkbox),
+	.d2l-table > * > tr > :has(d2l-selection-select-all),
+	.d2l-table > * > tr > :has(d2l-input-checkbox),
+	.d2l-table > * > tr > :has(d2l-selection-input) {
+		padding-block: calc(0.5 * (var(--d2l-table-cell-overall-height) - ${cssSizes.inputBoxSize}rem));
+	}
+	.d2l-table > * > tr.d2l-table-selected-first > :has(.d2l-checkbox),
+	.d2l-table > * > tr.d2l-table-selected-first > :has(d2l-input-checkbox),
+	.d2l-table > * > tr.d2l-table-selected-first > :has(d2l-selection-input) {
+		padding-bottom: calc(0.5 * (var(--d2l-table-cell-overall-height) - ${cssSizes.inputBoxSize}rem));
+		padding-top: calc(0.5 * (var(--d2l-table-cell-overall-height) - ${cssSizes.inputBoxSize}rem) - 1px);
+	}
+	.d2l-table > * > tr > :has(d2l-button-icon) {
+		padding-block: calc(0.5 * (var(--d2l-table-cell-overall-height) - 2rem - 2px)) !important;
+	}
+	@supports not selector(:has(a, b)) {
+		.d2l-checkbox,
+		d2l-input-checkbox,
+		d2l-selection-select-all,
+		d2l-selection-input {
+			margin-block: calc(0.5 * (var(--d2l-table-cell-height) - ${cssSizes.inputBoxSize}rem));
+		}
+		.d2l-table > * > tr.d2l-table-selected-first d2l-input-checkbox,
+		.d2l-table > * > tr.d2l-table-selected-first d2l-selection-input,
+		.d2l-table > * > tr.d2l-table-selected-first .d2l-checkbox {
+			margin-bottom: calc(0.5 * (var(--d2l-table-cell-height) - ${cssSizes.inputBoxSize}rem));
+			margin-top: calc(0.5 * (var(--d2l-table-cell-height) - ${cssSizes.inputBoxSize}rem) - 1px);
+		}
+		d2l-button-icon {
+			margin-block: calc(0.5 * (var(--d2l-table-cell-height) - var(--d2l-button-icon-min-height, 42px)));
+		}
+		d2l-table-wrapper[type="light"] .d2l-table d2l-button-icon {
+			margin-bottom: calc(0.5 * (var(--d2l-table-cell-height) - var(--d2l-button-icon-min-height, 42px)));
+			margin-top: calc(0.5 * (var(--d2l-table-cell-height) - var(--d2l-button-icon-min-height, 42px)) + 1px);
+		}
+		.d2l-table-header-col-sortable-siblings d2l-button-icon {
+			margin-block: calc(0.5 * (var(--d2l-table-cell-overall-height) - 2 * var(--d2l-table-cell-col-sort-button-has-siblings-cell-padding) - var(--d2l-button-icon-min-height, 42px)));
+		}
+		d2l-table-wrapper[type="light"] .d2l-table-header-col-sortable-siblings d2l-button-icon {
+			margin-bottom: calc(0.5 * (var(--d2l-table-cell-overall-height) - 2 * var(--d2l-table-cell-col-sort-button-has-siblings-cell-padding) - var(--d2l-button-icon-min-height, 42px)) + 2px);
+			margin-top: calc(0.5 * (var(--d2l-table-cell-overall-height) - 2 * var(--d2l-table-cell-col-sort-button-has-siblings-cell-padding) - var(--d2l-button-icon-min-height, 42px)) + 3px);
+
+		}
+	}
```

Comment:
Yeah, I changed the order so that the fallback would not depend on `selector` being supported. [CanIUse @supports selector](https://caniuse.com/mdn-css_at-rules_supports_selector). 

Since current browsers support `selector(...)`, I think I tested the behaviour of `not` when an unsupported at-rule is provided:
```css
.my-test {
     color: yellow;
}
@supports not selector2(:has(a, b)) {
     .my-test {
          color: green;
     }
}
```
I think I found that for Chrome, that the overall statement (`@supports not foo`) resolved to true, and the internal fallback CSS applied as we'd want. But for Safari, I think I found that it discarded the entire block, and the fallback styles not applied. It might be better to do a more specific test using `selector` and older browsers in Sauce, but I _think_ the above mimics the scenario.

Reply:
As for whether it would be ok to ignore that, I think depends on the case. If it's purely some aesthetic ugliness on super old browser, we're often "ok" with it when the it avoids complex hacks. I chose to flip the order because I was able to do-so, and it would be an a11y concern if the fallback didn't work.

Reply:
Thanks for that info - looks like I can flip it around and it will still looks pretty much as expected (the vdiffs have super slight variance that I'm not worried about), so I'll do that.